src/app/plugins/platform/Docker/plugins/cms/Drupal8/installDrupal.ts

```diff
@@ -120,4 +114,84 @@ async function installDrupal({
   }
 }
 
+/**
+ * Rename the generated web root if necessary.
+ *
+ * @param {InstallDrupalOptions} {
+ *     serviceDirectory,
+ *     documentRoot,
+ *   }
+ * @returns {Promise<string>} - The path to the Drupal installation root.
+ */
+async function renameWebRoot({
+  serviceDirectory,
+  documentRoot,
+}: InstallDrupalOptions): Promise<string> {
+  const drupalRoot = path.join(serviceDirectory, 'drupal');
+
+  // Rename 'web' in generated files, if we need to.
+  if (documentRoot !== 'web') {
+    await Promise.all([
+      replaceIn(path.join(drupalRoot, 'composer.json'), /web\//g, documentRoot + '/'),
+      replaceIn(path.join(drupalRoot, '.gitignore'), /web\//g, documentRoot + '/'),
+      replaceIn(path.join(drupalRoot, 'scripts/composer/ScriptHandler.php'), /\/web/g, '/' + documentRoot),
+      replaceIn(path.join(drupalRoot, 'README.md'), /web/g, documentRoot),
+    ]);
+
+    // Composer create-project now "helpfully" installs the web/ directory for us, which
+    // we don't want when it isn't the document root.
+    const webRoot = path.join(drupalRoot, 'web');
+    await rimrafAsync(webRoot);
+  }
+
+  return drupalRoot;
+}
+
+/**
+ * Confirm a valid web root selction was made for the selected platform.
+ *
+ * @param InstallDrupalOptions - Project installation options for validation.
+ *
+ * @throws Throws an error if an invalid web root name was selected.
+ */
+async function validateWebRoot({
+  projectType,
+  documentRoot,
+}: InstallDrupalOptions): Promise<void> {
+  // Crash early if the user asked for a non-'web' root for a Pantheon project.
+  // This will help prevent a lot of headaches due to misalignment with the platform's
+  // requirements.
+  if (projectType === pantheonProject && documentRoot !== 'web') {
+    throw new Error(
+      `Pantheon projects do not support '${documentRoot}' as the document root.`,
+    );
+  }
+}
+
+/**
+ * Use Composer to scaffold the service and project directory.
+ *
+ * @param {InstallDrupalOptions} {
+ *   serviceDirectory,
+ *   projectType,
+ * }
+ * @returns {Promise<void>}
+ */
+async function scaffoldServiceDirectory({
```

Comment:
As an organizational nit - prefer ordering things so that functions are declared before they're used. In this file's case, it means that these new functions should go before the `installDrupal` declaration.